### DISCUSSION TITLE
Does not work with bun?

### DISCUSSION THREAD

**Discussion by BennyAlex** [typicode/husky#1430] [Q&A]
I use husky with bun (installed in WSL) and always seeing this error when starting the server:

![image](https://github.com/typicode/husky/assets/17402724/067f22c4-b04b-4ac5-a0c8-fa7ddbefbc00)

I this a problem with bun + husky?

**Comment by BennyAlex** [comment]
maybe we can have a fix just using `o.scripts = o.scripts || {}` instead?

**Comment by typicode** [comment]
I'm suprised, it's not a new syntax :thinking:

**Reply by BennyAlex** [reply]
Lookd like husky is always running a script with node? So I only have an older version installed.

Would be good if husky checks for both bun and node I guess

**Reply by typicode** [reply]
I don't know of a CLI doing that. Maybe possible but I doubt it.
You could maybe alias node to bun.

**Comment by ignaciobockl** [comment]
Hi @BennyAlex 

Husky v9+ supports Bun natively. The issue is likely that your prepare script in package.json is using npx instead of bunx.

Install & Init
bun add --dev husky bunx husky init

This creates .husky/pre-commit and updates your prepare script.

Fix your package.json
Make sure your prepare script uses Bun:

json { "scripts": { "prepare": "bun run husky" } }

If your prepare script still says npx husky or husky install, that's the problem — Bun's bunx is the equivalent of npx. Replace it and run bun install again.

On older Husky versions (< v9)
The init command may not be available for Bun. Upgrade first:

bun add --dev husky@latest bunx husky init

The hooks themselves (.husky/pre-commit, etc.) are plain shell scripts — they work the same regardless of whether you use npm, pnpm, or Bun as your package manager.
